src/Orleans/CodeGeneration/TypeUtils.cs

```diff
@@ -561,7 +561,7 @@ public static IEnumerable<TypeInfo> GetDefinedTypes(Assembly assembly, Logger lo
             }
             catch (Exception exception)
```

Comment:
The intention was to keep the same behavior. Should non-`ReflectionTypeLoadException` exceptions bubble up?